aries_vcx/src/protocols/issuance_v2/formats/holder/hyperledger_indy.rs

```diff
@@ -0,0 +1,240 @@
+        MaybeKnown::Known(ProposeCredentialAttachmentFormatType::HyperledgerIndyCredentialFilter2_0)
+    }
+    fn get_request_attachment_format() -> MaybeKnown<RequestCredentialAttachmentFormatType> {
+        MaybeKnown::Known(
+            RequestCredentialAttachmentFormatType::HyperledgerIndyCredentialRequest2_0,
+        )
+    }
+    fn get_offer_attachment_format() -> MaybeKnown<OfferCredentialAttachmentFormatType> {
+        MaybeKnown::Known(OfferCredentialAttachmentFormatType::HyperledgerIndyCredentialAbstract2_0)
+    }
+    fn get_credential_attachment_format() -> MaybeKnown<IssueCredentialAttachmentFormatType> {
+        MaybeKnown::Known(IssueCredentialAttachmentFormatType::HyperledgerIndyCredential2_0)
+    }
+
+    async fn create_proposal_attachment_content(
+        data: &HyperledgerIndyCreateProposalInput,
+    ) -> VcxResult<Vec<u8>> {
+        let filter_bytes = serde_json::to_vec(&data.cred_filter)?;
+
+        Ok(filter_bytes)
+    }
+
+    fn extract_offer_details(
+        offer_message: &OfferCredentialV2,
+    ) -> VcxResult<HyperledgerIndyOfferDetails> {
+        let attachment = Self::extract_offer_attachment_content(offer_message)?;
+
+        Ok(serde_json::from_slice(&attachment)?)
+    }
+
+    async fn create_request_attachment_content(
+        offer_message: &OfferCredentialV2,
+        data: &Self::CreateRequestInput,
+    ) -> VcxResult<(Vec<u8>, HyperledgerIndyCreatedRequestMetadata)> {
+        let offer_bytes = Self::extract_offer_attachment_content(&offer_message)?;
+        let offer_payload = String::from_utf8(offer_bytes).map_err(|_| {
+            AriesVcxError::from_msg(
+                AriesVcxErrorKind::EncodeError,
+                "Expected payload to be a utf8 string",
+            )
+        })?;
+
+        let cred_def_id = parse_cred_def_id_from_cred_offer(&offer_payload)?;
+        let entropy = &data.my_pairwise_did;
+        let ledger = data.ledger;
+        let anoncreds = data.anoncreds;
+
+        let (credential_request, credential_request_metadata, _, credential_def_json) =
+            create_anoncreds_credential_request(
+                ledger,
+                anoncreds,
+                &cred_def_id,
+                &entropy,
+                &offer_payload,
+            )
+            .await?;
+
+        Ok((
+            credential_request.into(),
+            HyperledgerIndyCreatedRequestMetadata {
+                credential_request_metadata,
+                credential_def_json,
+            },
+        ))
+    }
+
+    async fn create_request_attachment_content_independent_of_offer(
+        _: &Self::CreateRequestInput,
+    ) -> VcxResult<(Vec<u8>, Self::CreatedRequestMetadata)> {
+        Err(AriesVcxError::from_msg(
+            AriesVcxErrorKind::ActionNotSupported,
+            "Anoncreds cannot create request payload independent of an offer",
+        ))
+    }
+
+    async fn process_and_store_credential(
```

Comment:
Assuming we would keep track of the `credential-request` we have sent, we could also validate that the the received credential is matching credential request we have sent. 

Afaik AFJ is doing this; Not a hard requirement, but perhaps nice2have feature down the line.

Reply:
yea agreed

Reply:
tracking this in issue https://github.com/hyperledger/aries-vcx/issues/1044